lib/common/appstart/app_initializer.dart

```diff
@@ -55,6 +66,52 @@ Future<void> initializeApp(bool isBackground) async {
     setup.setupScheduleSource();
   }
 
+  // Callback-Function for synchronizing the device calendar with the schedule, when schedule is updated
+  var preferenceProvider = KiwiContainer().resolve<PreferencesProvider>();
+  KiwiContainer()
+      .resolve<ScheduleProvider>()
+      .addScheduleUpdatedCallback((schedule, start, end) async {
+    List<DateEntry> listDateEntries = List<DateEntry>.empty(growable: true);
+    schedule.entries.forEach(
+      (element) {
+        DateEntry date = DateEntry(
+            room: element.room,
+            comment: element.details,
+            databaseName: 'DHBW',
+            description: element.title,
+            year: element.start.year.toString(),
+            start: element.start,
+            end: element.end);
+        listDateEntries.add(date);
+      },
+    );
+    KiwiContainer().resolve<ListDateEntries30d>().listDateEntries =
+        listDateEntries;
+
+    if (await preferenceProvider.isCalendarSyncEnabled()) {
+      Calendar selectedCalendar = await preferenceProvider.getSelectedCalendar();
+      if(selectedCalendar == null) return;
+      CalendarAccess().addOrUpdateDates(listDateEntries, selectedCalendar);
+    }
+  });
```

Comment:
Der Code an sich ist super.
Bitte lager diesen Code aber noch in eine separate Klasse aus, die dann so ungefähr aufgerufen wird:

```dart
var synchronizer = CalendarSynchronizer(
    KiwiContainer().resolve<ScheduleProvider>(),
    KiwiContainer().resolve<ScheduleSourceProvider>()
);

synchronizer.registerSynchronizationCallback() // <-- Hier ist der Code von Zeile 73 - 95
synchronizer.scheduleSyncInAFewSeconds() // <-- Hier der Code von Zeile 99 - 108
```